xbmc/pvr/channels/PVRChannelGroupMergedByName.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PVRChannelGroupMergedByName.h"
+
+#include "guilib/LocalizeStrings.h"
+#include "pvr/channels/PVRChannelGroupMember.h"
+#include "utils/StringUtils.h"
+
+#include <unordered_map>
+#include <utility>
+
+using namespace PVR;
+
+bool CPVRChannelGroupMergedByName::ShouldBeIgnored(
+    const std::vector<std::shared_ptr<CPVRChannelGroup>>& allChannelGroups) const
+{
+  std::unique_lock<CCriticalSection> lock(m_critSection);
+
+  // Ignore the group if it is empty or only members from one group are present.
+
+  if (m_members.empty())
+    return true;
+
+  const std::string mergedGroupName{!ClientGroupName().empty() ? ClientGroupName() : GroupName()};
+  int matchingGroup{-1};
+
+  for (const auto& member : m_members)
+  {
+    for (const auto& group : allChannelGroups)
+    {
+      if (group->GetOrigin() != Origin::USER && group->GetOrigin() != Origin::CLIENT)
+        continue;
+
+      if (group->GroupName() != mergedGroupName && group->ClientGroupName() != mergedGroupName)
+        continue;
+
+      if (group->IsGroupMember(member.second))
+      {
+        if (matchingGroup != -1 && matchingGroup != group->GroupID())
+        {
+          // Found a second group containing a member of this group. This group must not be ignored.
+          return false;
+        }
+        // First match or no new group. Continue with next group.
+        matchingGroup = group->GroupID();
+      }
+    }
+  }
+  return true;
+}
+
+namespace
+{
+std::vector<std::string> GetGroupNames(const std::vector<std::shared_ptr<CPVRChannelGroup>>& groups)
+{
+  std::unordered_map<std::string, unsigned int> knownNames;
```

Comment:
Suggest renaming variable, maybe `knownNamesCountMap` so it's clear it's purpose when reading the code.